src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -0,0 +1,386 @@
+            promise.resolve();
+        });
+        return promise;
+    };
+
+    /**
+     * Retrieves a promise for the appropriate hyphenator. If a hyphenator has not already been created, it will attempt
+     * to create one and assign the related promise to the `hyphenators` member for future retrieval.
+     *
+     * When creating a hyphenator, it first checks if there is configuration for the specified `lang`. If that fails,
+     * it attempts to fall back to a less specific localization.
+     *
+     * @param {Component} that - an instance of `fluid.prefs.enactor.syllabification`
+     * @param {String} lang - a valid BCP 47 language code. (NOTE: supported lang codes are defined in the
+     *                        `patterns`) option.
+     *
+     * @return {Promise} - returns a promise. If a hyphenator is successfully created, it is resolved with it.
+     *                     Otherwise, it resolves with undefined.
+     */
+    fluid.prefs.enactor.syllabification.getHyphenator = function (that, lang) {
+        var promise = fluid.promise();
+        if (!lang) {
+            promise.resolve();
+            return promise;
+        }
+        lang = lang.toLowerCase();
+
+        // Use an existing hyphenator if available
+        var existing = that.hyphenators[lang];
+        if (existing) {
+            return existing;
+        }
+
+        // Attempt to create an appropriate hyphenator
+        var hyphenatorPromise;
+        var pattern = that.options.patterns[lang];
+
+        if (pattern) {
+            hyphenatorPromise = fluid.prefs.enactor.syllabification.createHyphenator(that, pattern, lang);
+            fluid.promise.follow(hyphenatorPromise, promise);
+            that.hyphenators[lang] = hyphenatorPromise;
+            return promise;
+        }
+
+        var langSegs = lang.split("-");
+        pattern = that.options.patterns[langSegs[0]];
+
+        if (pattern) {
+            hyphenatorPromise = fluid.prefs.enactor.syllabification.createHyphenator(that, pattern, langSegs[0]);
+            fluid.promise.follow(hyphenatorPromise, promise);
+        } else {
+            hyphenatorPromise = promise;
+            // If there no available patterns to match the specified language, resolve the promise with out a
+            // hyphenator (undefined).
+            promise.resolve();
+        }
+
+        that.hyphenators[lang] = hyphenatorPromise;
+        that.hyphenators[langSegs[0]] = hyphenatorPromise;
+        return promise;
+    };
+
+    fluid.prefs.enactor.syllabification.syllabify = function (that, node, lang) {
+        var hyphenatorPromise = that.getHyphenator(lang);
+        hyphenatorPromise.then(function (hyphenator) {
+            if (hyphenator) {
+                // remove \u200B characters added hyphenateText
+                var hyphenated = hyphenator.hyphenateText(node.textContent).replace(/\u200B/gi, "");
```

Comment:
This is too vital a part of policy to be hidden away inside a promise callback. The body should be a public function, possibly with a bit of paramterisation